packages/relay/src/lib/services/clientService/index.ts

```diff
@@ -0,0 +1,276 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import dotenv from 'dotenv';
+import findConfig from 'find-config';
+import { AccountId, Client, PrivateKey } from '@hashgraph/sdk';
+import { Logger } from 'pino';
+import { Registry, Counter } from 'prom-client';
+import { SDKClient } from '../../clients/sdkClient';
+
+export default class ClientService {
+  private transactionCount: number;
+  private errorCount: number;
+  private resetDuration: number;
+  private shouldReset: boolean;
+
+  private isEnabled: boolean;
+  private isTimeResetDisabled: boolean;
+
+  private clientMain: Client;
+
+  /**
+   * The SDK Client use for connecting to both the consensus nodes and mirror node. The account
+   * associated with this client will pay for all operations on the main network.
+   *
+   * @private
+   */
+  private client: SDKClient;
+
+  /**
+   * The logger used for logging all output from this class.
+   * @private
+   */
+  private logger: Logger;
+  private readonly register: Registry;
+  private clientResetCounter: Counter;
+
+  private chainIds = {
+    mainnet: 0x127,
+    testnet: 0x128,
+    previewnet: 0x129,
+  };
+
+  /**
+   * @param {Logger} logger
+   * @param {Registry} register
+   */
+  constructor(logger: Logger, register: Registry) {
+    dotenv.config({ path: findConfig('.env') || '' });
+
+    this.logger = logger;
+
+    const hederaNetwork: string = (process.env.HEDERA_NETWORK || '{}').toLowerCase();
+    this.clientMain = this.initClient(logger, hederaNetwork);
+    this.client = this.initSDKClient(logger, register);
+
+    const currentDateNow = Date.now();
+    this.transactionCount = parseInt(process.env.CLIENT_TRANSACTION_RESET!) || 0;
+    this.resetDuration = currentDateNow + parseInt(process.env.CLIENT_DURATION_RESET!) || 0;
+    this.errorCount = parseInt(process.env.CLIENT_ERROR_RESET!) || 0;
+    this.isEnabled = true;
+    this.isTimeResetDisabled = this.resetDuration === currentDateNow;
+
+    if (this.transactionCount === 0 && this.errorCount === 0 && this.isTimeResetDisabled) {
+      this.isEnabled = false;
+    }
+    this.shouldReset = false;
+
+    this.register = register;
+
+    const metricCounterName = 'rpc_relay_client_service';
+    register.removeSingleMetric(metricCounterName);
+    this.clientResetCounter = new Counter({
+      name: metricCounterName,
+      help: 'Relay Client Service',
+      registers: [register],
+      labelNames: ['transactions', 'duration', 'errors'],
+    });
+  }
+
+  /**
+   *  Decrement transaction counter. If 0 is reached, reset the client. Check also if resetDuration has been reached and reset the client, if yes.
+   */
+  private decrementTransactionCounter() {
```

Comment:
Q: Does this method need a check for isEnabled?

Reply:
No need, because it's a private method and we use it only when before returning the sdk client in another function and on the top of that function we already check isEnabled.